pedpy/methods/profile_calculator.py

```diff
@@ -383,57 +472,86 @@ def compute_speed_profile(
+        data: A pandas DataFrame containing `frame` and pedestrian `speed`
+            (result from :func:`~speed_calculator.compute_individual_speed`).
+            Depending on `speed_method`, additional columns `x`, `y`, or
+            `polygon` might be required. `polygon` column
             (from :func:`~method_utils.compute_individual_voronoi_polygons`)
-            when using the `DensityMethod.VORONOI`. When computing the classic
-            density profile (`DensityMethod.CLASSIC`) the DataFrame needs to
-            contain the columns 'x' and 'y'. Computing the speed
-            profiles needs a `polygon` column (from
-            :func:`~method_utils.compute_individual_voronoi_polygons`) when
-            using the `SpeedMethod.VORONOI` or `SpeedMethod.ARITHMETIC`.
+            is required when using the :attr:`SpeedMethod.VORONOI` or
+            :attr:`SpeedMethod.ARITHMETIC`.
+            When computing the Gaussian profile (:attr:`SpeedMethod.GAUSSIAN`)
+            the DataFrame needs to contain the columns `x` and `y`.
             For getting a DataFrame containing all the needed data, you can
-            merge the results of the different function on the 'id' and
-            'frame' columns (see :meth:`pandas.DataFrame.merge` and
+            merge the results of the different function on the `id` and
+            `frame` columns (see :func:`pandas.DataFrame.merge` and
             :func:`pandas.merge`).
-        walkable_area (WalkableArea): geometry for which the profiles are
-            computed
-        grid_size: resolution of the grid used for computing the
-            profiles
-        speed_method: speed method used to compute the
+        walkable_area: geometry for which the speed profiles are
+            computed.
+        grid_size: The resolution of the grid used for computing the
+            profiles, expressed in the same units as the `walkable_area`.
+        speed_method: The speed method used to compute the
             speed profile
-        grid_intersections_area: intersection of grid cells with the Voronoi
+        grid_intersections_area: (Optional) intersection areas of grid cells
+            with Voronoi polygons, required for some speed methods.
             polygons (result from
             :func:`compute_grid_cell_polygon_intersection_area`)
-        fill_value: fill value for cells with no pedestrians inside when using
-            `SpeedMethod.MEAN` (default = `np.nan`)
+        fill_value: fill value for cells with no pedestrians inside when
+            using :attr:`SpeedMethod.MEAN` (default = `np.nan`)
+        gaussian_width: (Optional) The full width at half maximum (FWHM) for
+            Gaussian weights, required when using :attr:`SpeedMethod.GAUSSIAN`
+            (default = 0.5).
 
     Returns:
-        List of speed profiles
+        A list of NumPy arrays, each representing the speed profile for a
+        different grid cell within the walkable area.
```

Comment:
From my point of view a little bit misleading. Maybe something like:
A list of 2d-NumPy arrays, each representing the speed profile per frame.